src/storage/src/hummock/local_version_manager.rs

```diff
@@ -511,21 +508,58 @@ impl LocalVersionManager {
         for result in join_all(join_handles).await {
             result.expect("should be able to join the flush handle")
         }
-        let (order_index, task_payload, task_write_batch_size) = match self
-            .local_version
-            .write()
-            .get_mut_shared_buffer(epoch)
-            .and_then(|shared_buffer| shared_buffer.new_upload_task(SyncEpoch))
-        {
-            Some(task) => task,
-            None => {
-                tracing::trace!("sync epoch {} has no more task to do", epoch);
-                return Ok(0);
+        let notify1 = self.sync_epoch_notify.clone();
+        // TODO(xuxinhao): modify it after supporting uploading multiple shared buffers.#4442
+        loop {
+            let epoch1 = self
+                .local_version
+                .read()
+                .iter_shared_buffer()
+                .find(|(key, _)| key > &&self.max_sync_epoch.read().clone())
+                .map(|(key, _)| key)
+                .cloned();
+            if epoch1.is_none() || epoch1.unwrap() == epoch {
```

Comment:
Potential deadlock if we have an epoch with no writes.

For example, we have epoch1, epoch2, epoch3. If there are no writes in epoch2, we won't have an epoch2 entry in shared buffer. Therefore, `sync_shared_buffer(epoch2)` will be blocked forever.